### PR TITLE
Update safe spawn API to use new world height bounds

### PR DESCRIPTION
**Pull request by jasonw4331**
## Introduction

`World::getSafeSpawn()` operates with the expectation the y is above 0 for some undocumented reason. For worlds which generate below Y level 0, like some flat worlds, this makes the API ignore any passed parameters. 

### API changes

`World::getSafeSpawn()` should be mentioned in the changelog with support for lower world heights

### Behavioral changes

`World::getSafeSpawn()` now accepts Vector3 objects with the Y param as low as the world minimum

## Backwards compatibility

N/A

## Follow-up

It should be considered whether to test if the max y value should also be checked by this function

## Tests

TODO

### CODE REVIEW

File: src/world/World.php [hunk 1]
```diff
@@ -2951,7 +2951,7 @@ function() use ($resolver) : void{
 	 * @throws WorldException if the terrain is not generated
 	 */
 	public function getSafeSpawn(?Vector3 $spawn = null) : Position{
-		if(!($spawn instanceof Vector3) || $spawn->y < 1){
+		if($spawn === null || $spawn->y < $this->minY){
```
Comment (JavierLeon9966) [inline src/world/World.php]:
```suggestion
		if($spawn === null || $spawn->y < $this->minY + 1){
```
There must be at least one block where the player can stand otherwise, the player will spawn above the void.

**Review by JavierLeon9966 — CHANGES_REQUESTED** [review]

File: src/world/World.php (hunk 1 above)
Comment (dktapps) [reply src/world/World.php]: <= can be used instead

Comment (dktapps) [inline src/world/World.php]: This instanceof change seems unrelated

**Review by dktapps — CHANGES_REQUESTED** [review]

**Review by dktapps — APPROVED** [review]
